src/Symfony/Component/Security/Core/Authorization/Voter/ClosureVoter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authentication\AuthenticationTrustResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+
+/**
+ * This voter allows using a closure as the attribute being voted on.
+ * The following arguments are passed to the closure, in this order:
```

Comment:
We don't have to be order-dependent if we call the closure using a `...$vars`:
```php
function f($a, $b, ...$vars) {
}

f(...['a' => 1, 'b' => 2, 'c' => 3]);
```

Reply:
Would this work if the closure does *not* have a variadic argument to collect unknown arguments ? Forcing users to have this argument would be weird.

Reply:
Indeed, declaring $vars is going to be required. But the way I see it, this might improve discovery, so DX also.

Reply:
Is this really improving DX ? Forcing such signatures collecting parameters by name only while forcing the variadic argument collecting extra ones is not common at all. Users will likely be confused about it and tooling will tell them to remove the unused argument `$vars`.
And I don't see how this helps discovery either as they still won't know what they can reference (and static analysis tools and IDEs won't help them discover this either, as it will be impossible to describe the signature)

Reply:
OK, so my suggestion can be updated this way: let's use named parameters when calling the closure.
That will make the name of the arguments part of the contract, thus breaking the order requirement (and still allowing my variadic trick to make the signature shorter on the closure side).

Reply:
That's updated. I just have [this question](https://github.com/symfony/symfony/pull/59150#discussion_r1952159315) which is about how to emphasize named arguments matters in this case.

Reply:
They can use a variadic as I mentionned: `fn (...$vars) => true`

Reply:
the fact that they must define all arguments (or a variadic one) also means that we can never add more arguments in the signature as that would be a BC break for projects that don't have the variadic arg at the end. Do we really want to impose us such restriction forever ?

Reply:
That day, we'll be able to use reflection to decide which style works. (we could do it today but I'm not sure it's worth it)